service/src/main/java/tds/assessment/services/impl/AssessmentServiceImpl.java

```diff
@@ -47,32 +48,13 @@ public AssessmentServiceImpl(AssessmentQueryRepository assessmentQueryRepository
             List<ItemProperty> itemProperties = itemQueryRepository.findActiveItemsProperties(assessmentKey);
             List<ItemConstraint> itemConstraints = assessmentQueryRepository.findItemConstraintsForAssessment(clientName,
                     assessment.getAssessmentId());
-            assessment.setItemConstraints(itemConstraints);
+            Set<Strand> strands = strandQueryRepository.findStrands(assessmentKey);
 
-            assignPropertiesToItems(items, itemProperties);
-            assignFormsAndItemsToAssessmentSegments(assessment, forms, items);
+            AssessmentAssembler.assignPropertiesToItems(items, itemProperties);
```

Comment:
I would combine all these calls into a single call.  Pass in all the items then have it return the assessment.  I know you'll be updating it via reference but I think it'd read clearer if it was something like `assessment = assessmentAssembler.assemble(data...);`